File: Golang/SingleNumber/SingleNumber_test.go
package SingleNumber

import "testing"

func TestSingleNumber(t *testing.T) {

	if singleNumber(nil) == 0 {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if singleNumber([]int{1}) == 1 {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if singleNumber([]int{1, 1, 2}) == 1 {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if singleNumber([]int{1, 1, 2, 2, 3}) == 1 {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

}

File: Python/SingleNumberTest.py
from unittest import TestCase

from SingleNumber import SingleNumber


class TestSingleNumber(TestCase):
    def test_singleNumber(self):
        sn = SingleNumber()
        self.assertTrue(sn.singleNumber(None) == 0)

        self.assertTrue(sn.singleNumber([1]) == 1)

        self.assertTrue(sn.singleNumber([1, 1, 2]) == 2)

        self.assertTrue(sn.singleNumber([1, 1, 2, 2, 3]) == 3)
